src/main/java/com/contentgrid/gateway/security/jwt/issuer/PropertiesBasedJwtClaimsSigner.java

```diff
@@ -4,89 +4,59 @@
 import com.nimbusds.jose.JOSEObjectType;
 import com.nimbusds.jose.JWSAlgorithm;
 import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.JWSSigner;
 import com.nimbusds.jose.crypto.factories.DefaultJWSSignerFactory;
-import com.nimbusds.jose.jwk.ECKey;
 import com.nimbusds.jose.jwk.JWK;
+import com.nimbusds.jose.jwk.JWKMatcher;
+import com.nimbusds.jose.jwk.JWKSelector;
 import com.nimbusds.jose.jwk.JWKSet;
 import com.nimbusds.jose.jwk.KeyUse;
-import com.nimbusds.jose.jwk.OctetKeyPair;
-import com.nimbusds.jose.jwk.RSAKey;
+import com.nimbusds.jose.jwk.source.JWKSource;
+import com.nimbusds.jose.proc.SecurityContext;
+import com.nimbusds.jose.proc.SimpleSecurityContext;
 import com.nimbusds.jose.produce.JWSSignerFactory;
 import com.nimbusds.jwt.JWTClaimsSet;
 import com.nimbusds.jwt.SignedJWT;
-import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.Collections;
+import java.util.Date;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Random;
 import java.util.Set;
-import java.util.stream.Stream;
 import lombok.RequiredArgsConstructor;
 import lombok.SneakyThrows;
-import org.springframework.core.io.Resource;
-import org.springframework.core.io.support.ResourcePatternResolver;
+import org.springframework.util.ConcurrentLruCache;
+
 
 @RequiredArgsConstructor
 public class PropertiesBasedJwtClaimsSigner implements JwtClaimsSigner {
-    private final JwtClaimsSignerProperties properties;
-    private final ResourcePatternResolver resourcePatternResolver;
+
     private final JWSSignerFactory jwsSignerFactory;
     private final Random random;
+    private final JWKSource<SecurityContext> jwkSource;
+    private final Set<JWSAlgorithm> algorithms;
+
 
-    public PropertiesBasedJwtClaimsSigner(JwtClaimsSignerProperties properties, ResourcePatternResolver resourcePatternResolver) {
-        this(properties, resourcePatternResolver, new DefaultJWSSignerFactory(), new Random());
+    public PropertiesBasedJwtClaimsSigner(JWKSource<SecurityContext> jwkSource, Set<JWSAlgorithm> algorithms) {
+        this(new DefaultJWSSignerFactory(), new Random(), jwkSource, algorithms);
     }
 
     public interface JwtClaimsSignerProperties {
```

Comment:
I think this interface does not belong here anymore, since it's not used by this class anymore.